### PR TITLE
add get_schema_operation_parameters() for Open API 3 generateschema

### LINKED ISSUE #1083: get_schema_operation_parameters()
As of current master branch (3.10 milestone) DRF `generateschema` expects each filter class to have a `get_schema_operation_parameters()` function which is defined in [rest_framework.filters.BaseFilterBackend](https://github.com/encode/django-rest-framework/blob/c2293e9f251b1f215825186a7bcbf5a006df0cb0/rest_framework/filters.py#L20-L38).

This is currently missing from `DjangoFilterBackend`  https://github.com/carltongibson/django-filter/blob/7c08deea132686f456df8a646a9c65654cee2712/django_filters/rest_framework/backends.py#L20

Perhaps update `DjangoFilterBackend` to extend `rest_framework.filters.BaseFilterBackend` as a start and then add an implementation?

### PR DESCRIPTION
**Pull request by n2ygk**
Fixes #1083 

Implements Open API 3 (formerly known as Swagger) schema generation by upstream DRF 3.9+ `generateschema` management command.

### CODE REVIEW

**Comment by carltongibson** [comment]
Hi @n2ygk. Super. Thanks for this.

File: django_filters/rest_framework/backends.py [hunk 1]
```diff
@@ -138,3 +138,25 @@ def get_schema_fields(self, view):
                 schema=self.get_coreschema_field(field)
             ) for field_name, field in filterset_class.base_filters.items()
         ]
+
+    def get_schema_operation_parameters(self, view):
+        try:
+            queryset = view.get_queryset()
```
[the remark this answers is not in the document]
Comment (carltongibson) [reply django_filters/rest_framework/backends.py]: Hi @n2ygk. I’m not sure what you mean. 

`get_queryset()` is canonical.

[the remark this answers is not in the document]
Comment (n2ygk) [reply django_filters/rest_framework/backends.py]: see this https://github.com/carltongibson/django-filter/pull/1086#pullrequestreview-247417781

**Comment by carltongibson** [comment]
“If I change...” 

Why would you do that? 🙂 

It’s just not right to access the queryset directly. 

(That’s what I don’t follow...)

**Comment by n2ygk** [comment]
> Why would you do that? 🙂

No good reason? Because it's there? Because it's named `queryset` not `_queryset` so it must be a "public" member? Because lots of bad examples out there like this: https://stackoverflow.com/questions/19707237/use-get-queryset-method-or-set-queryset-variable

**Comment by carltongibson** [comment]
I guess see [`rest_framework.generics`](https://github.com/encode/django-rest-framework/blob/809a6acd36b53017a8a41a1d46c816f0480cb20b/rest_framework/generics.py#L54-L56). 

tl;dr: Client code should always call `get_queryset()`.

**Comment by n2ygk** [comment]
> I guess see [`rest_framework.generics`](https://github.com/encode/django-rest-framework/blob/809a6acd36b53017a8a41a1d46c816f0480cb20b/rest_framework/generics.py#L54-L56).
> 
> tl;dr: Client code should always call `get_queryset()`.

Yep, learn something new every day.  Sometimes for the second or third time. :-)

**Comment by n2ygk** [comment]
@carltongibson just pinging to see if you are OK with merging this?
